### DISCUSSION TITLE
ConstraintViolationExceptions thrown when checked that user already in db

### DISCUSSION THREAD

**Discussion by Angel0926** [quarkusio/quarkus#36982] [Q&A]
Before I updated Quarkus to 3.2.7 I had no problems with ConstraintViolationExceptions being thrown (which eventually gets thrown as a persistanceException), since when there was a user in the database I had my own implementation of the error and it worked, but now at the end of the code when the user already exists, I throw a ConstraintViolationException. Are there any options for what the problem is?

**Comment by Angel0926** [comment]
@geoand @yrodiere

**Comment by geoand** [comment]
cc @gsmet

**Comment by yrodiere** [comment]
If I understand correctly:

* you're persisting a new instance of a "User" entity
* another instance of the "User" entity already exists in DB
* before Quarkus 3.2.7 you got a `PersistenceException` wrapping a `ConstraintViolationException`
* in Quarkus 3.2.7 you are getting a `ConstraintViolationException` directly

This seems like a change in Hibernate ORM, completely out of the control of Quarkus. If you feel that's a problem, I'd recommend you [reach out to the Hibernate ORM team](https://hibernate.org/community/), explaining why you don't think this should have changed.

If I did not understand correctly, I'm afraid I'll need a clearer explanation. A minimal reproducer -- a small Quarkus project that demonstrates the error -- would go a long way.

**Reply by yrodiere** [reply]
> This seems like a change in Hibernate ORM, completely out of the control of Quarkus

Actually I imagine this could also be a change in Panache, if you use Panache. But again, if I misundertood your problem, a reproducer would help.

**Reply by Angel0926** [reply]
@yrodiere  on versions 2xx, this request ended not with an exception, but with its implementation(UsersBatchUploadDto)  when there is a duplicate, now it crashes ConstraintViolationException and  is converted to persistance

**Reply by Angel0926** [reply]
@yrodiere  @geoand https://github.com/quarkusio/quarkus/issues/32578 i think this problem like mine

**Reply by yrodiere** [reply]
Seems you're using Hibernate Reactive, I'll let others more knowledgeable than me try to understand.

Two things before I leave:

1. Please use formatting for your code:
````
```
mycode
```
````
2. Please include that mysterious stacktrace you've been talking about.

**Reply by Angel0926** [reply]
what's you mean mysterious stacktrace?     i have this 
```

or if i dont have handle exception i get
```
 "details": "Error id 745e1fbf-d4d3-4fb4-922c-5c7f8faff392-8, org.hibernate.exception.ConstraintViolationException: error executing SQL statement [{errorMessage=Duplicate entry 'johnsmith' for key 'User.PRIMARY', errorCode=1062, sqlState=23000}] [insert into User (displayName,userLogin) values (?,?)]",
```

in versions 2.x.x my request did not reach this point and my answer was in the response(200 batchupload dto realisation), but now the request reaches this point

**Reply by Angel0926** [reply]
@yrodiere hi. do you have any news on this?

**Reply by yrodiere** [reply]
> what's you mean mysterious stacktrace? i have this

I mean you didn't include a stacktrace, so we don't know where the error originates from.

Though I guess with Reactive only the first line of the stacktrace is actually useful.


> or if i dont have handle exception i get
> 
> ```
>  "details": "Error id 745e1fbf-d4d3-4fb4-922c-5c7f8faff392-8, org.hibernate.exception.ConstraintViolationException: error executing SQL statement [{errorMessage=Duplicate entry 'johnsmith' for key 'User.PRIMARY', errorCode=1062, sqlState=23000}] [insert into User (displayName,userLogin) values (?,?)]",
> ```

That's a beginning, though the following lines starting with `at ` would help. AKA, the stack trace. They should appear somewhere in your logs, I guess?

> hi. do you have any news on this?

No, as I said:

> Seems you're using Hibernate Reactive, I'll let others more knowledgeable than me try to understand.

@DavideD may know more.

**Reply by DavideD** [reply]
Is this the same question asked in Zulip? https://hibernate.zulipchat.com/#narrow/stream/132096-hibernate-user/topic/ConstraintViolationExceptions

**Reply by DavideD** [reply]
One of the most common cause of mysterious errors is that some operations don't happen in the right order.
